packages/api/lambdas/replay-sqs-messages.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const pMap = require('p-map');
+
+const { envUtils } = require('@cumulus/common');
+const { getJsonS3Object, listS3ObjectsV2 } = require('@cumulus/aws-client/S3');
+const { getQueueUrlByName, sendSQSMessage } = require('@cumulus/aws-client/SQS');
+const { getS3PrefixForArchivedMessage } = require('@cumulus/ingest/sqs');
+const Logger = require('@cumulus/logger');
+
+const logger = new Logger({ sender: '@cumulus/replay-sqs-messages' });
+
+// Get messages from S3 using queueName
+const getArchivedMessagesFromQueue = async (queueName) => {
+  let continuationToken;
+  const validMessages = [];
+  const bucket = envUtils.getRequiredEnvVar('system_bucket', process.env);
+  const stackName = envUtils.getRequiredEnvVar('stackName', process.env);
+  const prefix = getS3PrefixForArchivedMessage(stackName, queueName);
+  const params = {
+    Bucket: bucket,
+    Prefix: prefix,
+    ContinuationToken: continuationToken,
+    Delimiter: '/',
+  };
+  logger.debug(`Params for listS3Keys bucket: ${bucket}, prefix: ${prefix}`);
+
+  const messageObjects = await listS3ObjectsV2(params);
+  const archivedMessages = await Promise.allSettled(messageObjects.map(
+    async (messageObject) => {
+      const sqsMessage = await getJsonS3Object(bucket, messageObject.Key);
+      logger.debug(`Message retrieved ${JSON.stringify(sqsMessage)}`);
+      return sqsMessage;
+    }
+  ));
+  archivedMessages.map((message) => {
+    if (message.status === 'fulfilled') {
+      validMessages.push(message.value);
+    }
+    return validMessages;
+  });
+  return validMessages;
```

Comment:
```suggestion
  const validMessages = archivedMessages
    .filter((message) => message.status === 'fulfilled')
    .map((message) => message.value);
  return validMessages;
```

I might change the variable name of `archivedMessage` to `archiveRequests` and `validMessages` to `archivedMessages` for clarity

Reply:
Ooh I didn't realize I could use map like how you've suggested! [`9cc6812` (#2373)](https://github.com/nasa/cumulus/pull/2373/commits/9cc6812cfe17335ca582246687deaed58fe6a3f0)